src/snyk/common/configuration/configuration.ts

```diff
@@ -184,9 +188,62 @@ export class Configuration implements IConfiguration {
     return `${authUrl.toString()}manage/snyk-code?from=vscode`;
   }
 
-  getSnykLanguageServerPath(): string {
-    if (!this.processEnv.SNYK_LS_PATH) throw new Error('SNYK_LS_PATH environment variable is not set.');
-    return this.processEnv.SNYK_LS_PATH;
+  getSnykLanguageServerPath(): string | undefined {
+    return this.workspace.getConfiguration<string>(
+      CONFIGURATION_IDENTIFIER,
+      this.getConfigName(ADVANCED_CUSTOM_LS_PATH),
+    );
+  }
+
+  getSnykLanguageServerPath2(): string {
+    // check and set platform
+    const platform = Platform.getCurrent();
+    if (!isPlatformSupported) {
+      throw new Error(`Platform ${platform} is not supported`);
+    }
+
+    // set default home directory
+    const homeDir = Platform.getHomeDir();
+
+    const lsPlatform = LsExecutable.getCurrentWithArch();
+
+    // set Language Server filename
+    const lsFilename = LsExecutable.filenameSuffixes[lsPlatform];
+
+    // set default Language Server path
+    const defaultPaths: Record<LsSupportedPlatform, string> = {
```

Comment:
Let's align with Eclipse implementation https://github.com/snyk/snyk-eclipse-plugin/blob/2d455fc827c7ba203b8999f358b62a0ae021a979/plugin/src/main/java/io/snyk/eclipse/plugin/utils/FileSystemUtil.java#L34

General tip: avoid having such lists as they're hard to maintain when new os/arch support being added.

Reply:
As suggested in Slack, I've aligned with the [Language Server implementation for the CLI](https://github.com/snyk/snyk-ls/blob/c7fad4a45c94bf76a23b05c0b0038f352155044f/application/config/config.go#L438).

It uses the [XDG package](https://pkg.go.dev/github.com/adrg/xdg@v0.4.0#section-readme).